### PR TITLE
Feature flag banner

### PR DESCRIPTION
**Pull request by nibivi77**
### Description
Adding conditionnal banner.

The goal is to have this displayed on a feature_flag. 

Couldn't figure out how to only display this when `NODE_ENV === 'prototype'`. Then again, maybe we want it in different environments so this could be okay.

### Related Azure Boards Work Items
[AB#3358](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3358)

### Screenshots (if applicable)
image

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions
`show-banner` must be added to the FEATURE_FLAGS env.

### CODE REVIEW

File: frontend/app/utils/env.server.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ function tryOrElseFalse(fn: () => unknown) {
 const validMockNames = ['cct', 'lookup', 'power-platform', 'raoidc', 'status-check', 'wsaddress', 'subscription'] as const;
 export type MockName = (typeof validMockNames)[number];
 
-const validFeatureNames = ['doc-upload', 'email-alerts', 'hcaptcha', 'view-personal-info', 'view-applications', 'view-letters', 'view-messages', 'edit-personal-info', 'status', 'authenticated-status-check'] as const;
+const validFeatureNames = ['doc-upload', 'email-alerts', 'hcaptcha', 'view-personal-info', 'view-applications', 'view-letters', 'view-messages', 'edit-personal-info', 'status', 'authenticated-status-check', 'show-banner'] as const;
```
Comment (sebastien-comeau) [inline frontend/app/utils/env.server.ts:25]: Change it to a more specific name `show-prototype-banner`

File: frontend/public/locales/en/gcweb.json [hunk 2]
```diff
@@ -29,7 +29,9 @@
     "menu-security-settings.href": "{{baseUri}}/en/security-settings",
     "menu-contact-us.text": "Contact Us",
     "menu-contact-us.href": "{{baseUri}}/en/contact-us",
-    "menu-sign-out.text": "Sign Out"
+    "menu-sign-out.text": "Sign Out",
+    "banner-alert": "Info",
+    "banner-desc": "This is a prototype environment. No data is kept."
```
Comment (sebastien-comeau) [inline frontend/public/locales/en/gcweb.json:34]: Change the names to a more specific name.

```suggestion
    "prototype-banner": {
      "alert": "Info",
      "desc": "This is a prototype environment. No data is kept."
    }
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/utils/env.server.ts (hunk 1 above)
Comment (nibivi77) [reply frontend/app/utils/env.server.ts:25]: That's the thing, we might want to use it for any other reason. Possibly?

File: frontend/app/components/banner.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ReactNode } from 'react';
+
+interface BannerProps {
+  alert?: ReactNode;
+  description?: ReactNode;
+}
+
+export function Banner({ alert, description }: BannerProps) {
+  return (
+    <div className="font-body bg-sky-600 text-white" data-testid="banner-id">
```
Comment (sebastien-comeau) [inline frontend/app/components/banner.tsx:10]: Use `bg-sky-800` to be closer to RH and Passport banner background.

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/utils/env.server.ts (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/utils/env.server.ts:25]: That feature flag name doesn't reflect the feature that is showing a prototype banner for the moment.

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**